importer/helper.go

```diff
@@ -252,17 +252,33 @@ func updateAccounting(db idb.IndexerDb, genesisJSONPath string, numRoundsLimit i
 	}
 
 	lastlog := time.Now()
-	act := accounting.New(db)
+	act := accounting.New()
 	txns := db.YieldTxns(context.Background(), state.AccountRound)
 	currentRound := uint64(0)
 	roundsSeen := 0
 	lastRoundsSeen := roundsSeen
+	var blockPtr *types.Block = nil
 	for txn := range txns {
 		maybeFail(txn.Error, l, "updateAccounting txn fetch, %v", txn.Error)
 		if txn.Round != currentRound {
+			if blockPtr != nil {
+				// TODO: commit rounds with empty paysets to avoid a special case to update the db metastate.
```

Comment:
I think with current logic metastate doesn't update when there are empty blocks, and this works for accounting but looks weird when following along trying to guess index db state from doing SQL queries.

Reply:
The special case I'm talking about in this comment is the HandleBlock function in `daemon.go`. It checks the payset size and updates the metastate, which seems really out of place. If `CommitRoundAccounting` were able to handle rounds with no transactions I think we could remove it:
```
func (bih *blockImporterHandler) HandleBlock(block *types.EncodedBlockCert) {
        ...
	bih.imp.ImportDecodedBlock(block)
	importer.UpdateAccounting(bih.db, genesisJSONPath, logger)
	dt := time.Now().Sub(start)
	if len(block.Block.Payset) == 0 {
		// update the meta state here!!
	}
	logger.Infof("round r=%d (%d txn) imported in %s", block.Block.Round, len(block.Block.Payset), dt.String())
	bih.round = uint64(block.Block.Round)
}
```